x-pack/plugins/fleet/common/types/models/agent.ts

```diff
@@ -443,7 +443,7 @@ export interface AgentUpgradeDetails {
   target_version: string;
   action_id: string;
   state: AgentUpgradeStateType;
-  metadata: {
+  metadata?: {
```

Comment:
Note: this is a fix to the upgrade details implementation (https://github.com/elastic/kibana/pull/167539) as the `metadata` field is only present for certain states.